include/seastar/http/client.hh

```diff
@@ -78,6 +78,16 @@ public:
      */
     input_stream<char> in(reply& rep);
 
+    /**
+     * \brief Skips the reply body (if any)
+     *
+     * Can be called if the caller doesn't need the body contents of the reply (i.e. --
+     * only needs status and/or headers).
+     *
+     * Mutually exclusive with the \ref in(reply& rep) one.
+     */
+    future<> skip_body(reply& rep);
```

Comment:
There's nothing wrong with this code, I just have my doubts whether it's a good idea to add a method that does nothing but take in() and call the skip_entire_stream() method on it. Why is it hard for the user to do that? With this new method, you need to document that the two methods are mutually exlusive, as you did. Wouldn't it be simpler if you just said that everyone should call in(), and if you want to ignore the result, use skip_entire_stream() on the result?

But feel free to convince me otherwise. I don't feel strongly about this.

Reply:
Well, I don't have strong requirement on this part either :) As I wrote above it's just a sugar that saves few lines of code for callers. Like util::skip_entire_stream thing itself, it's not a big deal to call inp.consume() for the stream the caller wants to skip, but having a short method doing it for you is just neat